### DISCUSSION TITLE
Batch process for Land Cover Change | How path two different raster at each cicle?

### DISCUSSION THREAD

**Discussion by efbustos** [semiautomaticgit/SemiAutomaticClassificationPlugin#264] [Q&A] [answered]
Hi Luca and Friends

I'm exploring the capability of SCP to batch a Land Cover Change process for a set of 'places', in order to compare the changes in land cover classes between an initial and future condition.

The filenames structure is: "ClimClass_TIME_Site.tif", where "TIME" can be "Hist" for initial land cover raster, and "Fut" for future land cover raster. This couple of files for each "Site". All files are in the same folder "ClimClass_All".

Tried different configurations (e.g. same filename for each site, but in different folder for "Hist" and "Fut" conditions), but the process doesn't work. Using the scrip showed below, the error message in QGIS is "Information [4] : Select a raster; raster is not loaded"

My main questions are:
1. It's possible call two different files at each iteration in a batch process as inputs in one function? 
2. It's possible in a "!for_file_in!" cycle, use different "!File!" depending on some filter (e.g. "Hist" or "Fut" string)?
3. In order to save each result with the extension "_change", its correct the syntax showed below?

Thanks in advance for your help and support for this development.

!working_dir!; 'C:/..../ClimClass_All/'

!for_file_in!;'!working_dir!', 'tif'

land_cover_change; reference_raster_path: '!file_directory!, Hist, tif'; new_raster_path : '!file_directory!, Fut, tif'; output_raster_path : '!file_directory!_change';

!end_for_file!

**Comment by semiautomaticgit** [comment, marked as answer]
Hi @efbustos ,

unfortunately the batch tool is not suitable for filtering all the required inputs.
You could consider using Python, and the new library [Remotior Sensus](https://github.com/semiautomaticgit/remotior_sensus) which also includes a tool for [Cros classification](https://remotior-sensus.readthedocs.io/en/latest/remotior_sensus.tools.cross_classification.html) which is basically the same as land cover change comparing all the input classes with the option `cross_matrix=True`.
With Python you could easily filter input files, or create lists of file to be processed.

**Reply by efbustos** [reply]
Thanks, @semiautomaticgit , for your answer. I'll explore the library that you mentioned.
A last detail. So, to be completely clear, the Land Cover Change tool in QGIS-SCP, isn't capable to be process at batch mode? 
This because always will be necessary two different raster (with different names, or store in different folders) to be correctly used in batch by this tool, or not? Thanks again for your work!

**Reply by semiautomaticgit** [reply]
You are very welcome!
Batch can perform land cover change tool, but you should provide the full path of input files.
